anatomy_of_variables/Anatomy of Variables.md

```diff
@@ -0,0 +1,229 @@
+# Anatomy of Variables
+This post intends to look into what variables stand for, starting from what they are to how they can be used.
+
+By the end of this post, you should be able to:
+
+ * Explain what variables are.
+ * Identify the different types of variables in ruby.
+ * Explain what makes one variable different from one another.
+
+
+We will start from what variables are in programming generally, then sight some examples in some programming languages and finish off on **Ruby Variables**
+
+## What are Variables?
+**Variables**, as defined by [wikipedia](https://en.wikipedia.org/wiki/Variable_(computer_science)) " are storage locations paired with an associated symbolic name, ***an identifier***, which contains some known or unknown quantity of information referred to as a value." In other words, a variable can be referred to as a ***container*** that holds values or information. The variable's name represents what information the variable contains. They are called variables because the represented information can change but the operations on the variable remain the same.
+
+Variables can be likened to Buckets where information can be kept while on the outside of the buckets are names. When referring to one of the buckets, we use the name of the bucket, not the data stored in the bucket.
+
+Variables are the same across all computer programming languages. What differentiates them is the way they are used and how they are been assigned.
+
+##Types of Variables
+There are two general types of variables. They are:
+
+* Local Variables
+* Global Variables
+
+**Local Variables**
+
+These are variables that are assigned within a function or a method. They can be used only within the method or function in which they have been assigned.
+
+```ruby
+def counter
+    count = 0
+    while count <= 10
+        puts count
+        count += 1
+    end
+end
+
+count #=> 0 1 2 3 4 5 6 7 8 9 10
+```
+
+This is a ruby method named `counter`. A local variable `count` was assigned in this method and it was used to print `0..10`. However if you call count outside this method, it would throw a `method or variable not defined` error.
+
+Here is the JavaScript version of the above code:
+
+```JavaScript
```

Comment:
There is no actual reason for writing the same code in `JS`. I thought of removing it too, since this is a ruby tutorial.